### PR TITLE
bluetooth: controller: Update slot reservation time calculation for ADV

### PR DESCRIPTION
**Pull request by erbr-ot**
Correct slot reservation time calc for legacy advertise
Add dependency on data lengths for advertise and scan response

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_adv.c [hunk 1]
```diff
@@ -654,25 +650,68 @@ u8_t ll_adv_enable(u8_t enable)
 	ARG_UNUSED(rl_idx);
 #endif /* CONFIG_BT_CTLR_PRIVACY */
 
-	interval = adv->interval;
-	chan_map = lll->chan_map;
-	chan_cnt = util_ones_count_get(&chan_map, sizeof(chan_map));
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	const u8_t phy = lll->phy_p;
+#else
+	const u8_t phy = 1;
```
Comment (mped-oticon) [inline subsys/bluetooth/controller/ll_sw/ull_adv.c]: "Normal advertise is always at 1Mbps"

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_vendor.h [hunk 2]
```diff
@@ -5,8 +5,18 @@
...
+/* Worst-case time margin needed after event end-time in the air
+ * (done/preempt race margin + power-down/chain delay)
+ */
+#define EVENT_OVERHEAD_END_US         40
 #define EVENT_JITTER_US               16
+
+#define EVENT_RX_JITTER_US(phy) 16    /* Radio Rx timing uncertainty */
+#define EVENT_RX_TO_US(phy) (((((phy)&0x03) + 4)*8 / BIT((((phy)&0x3)>>1))) + \
+				  EVENT_RX_JITTER_US(phy))
+#define EVENT_RX_TX_TURNARROUND(phy)  ((phy) == 1?100:((phy) == 2 ? 80:900))
```
Comment (mped-oticon) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_vendor.h]: 100,80,900 us: How sure are we (Oticon) about these (Nordic) timing constants?

**Review by mped-oticon — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_adv.c (hunk 1 above)
Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_adv.c]: Assuming this means suggestion to add a comment to this effect, I will do just that.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_vendor.h (hunk 2 above)
Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_vendor.h]: Not at all. I was not sure how to handle that aspect, so I chose to simply add the numbers that Oticon uses, and expected for Nordic to change these to be correct.

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 3]
```diff
@@ -437,14 +453,14 @@ struct pdu_data {
 
 	union {
 		struct pdu_data_llctrl llctrl;
-		u8_t                   lldata[0];
+		u8_t				lldata[0];
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/pdu.h]: No use of tabs other than beginning of a line.

**Review by cvinayak — COMMENTED** [review]
Need few days next week to complete review.

**Comment by mped-oticon** [comment]
@erbr-ot In case you haven't already noticed, it seems like you are missing a file for defining `TARGETA_SIZE`.

**Comment by erbr-ot** [comment]
@nashif CI is failing on boards that ought not be affected by the changes in this PR. Advice appreciated.

**Comment by erbr-ot** [comment]
@carlescufi CI appear to be failing on boards that ought not be affected by the changes in this PR. I'd much appreciate advice on this.

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 4]
```diff
@@ -15,7 +15,19 @@
...
+#define ADVA_SIZE               6
+#define SCANA_SIZE              6
+#define INITA_SIZE              6
+#define TARGETA_SIZE            6
+#define LLDATA_SIZE             22
+#define CRC_SIZE                3
+#define PREAMBLE_SIZE(phy)      (phy&0x3)
+#define LL_HEADER_SIZE(phy)     (PREAMBLE_SIZE(phy) + PDU_AC_LL_HEADER_SIZE \
+				  + ACCESS_ADDR_SIZE + CRC_SIZE)
+#define BYTES2US(bytes, phy)    ((bytes)*8/BIT((phy&0x1)>>1))
```
Comment (mped-oticon) [inline subsys/bluetooth/controller/ll_sw/pdu.h]: nit: We should use `PKT_US` only.
`PKT_US` is now available for all to use (was limited to Nordic at first), which turns `BYTES2US` into duplicate code.

**Review by mped-oticon — CHANGES_REQUESTED** [review]
LGTM, but have one nitpick.

**Review by cvinayak — CHANGES_REQUESTED** [review]
Could you revert the whitespace changes, so that only actual change is under review.

File: subsys/bluetooth/controller/ll_sw/pdu.h (hunk 4 above)
Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/pdu.h]: PKT_US() macro is local to ull_conn.c and could possibly be modified to utilize BYTES2US.

**Comment by erbr-ot** [comment]
@cvinayak and @mped-oticon would you guys re-review, as I believe I have covered the issues mentioned.

**Review by mped-oticon — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_vendor.h [hunk 5]
```diff
@@ -9,4 +9,16 @@
...
 #define EVENT_OVERHEAD_PREEMPT_MAX_US EVENT_OVERHEAD_XTAL_US
 #define EVENT_OVERHEAD_START_US       200
+/* Worst-case time margin needed after event end-time in the air
+ * (done/preempt race margin + power-down/chain delay)
+ */
+#define EVENT_OVERHEAD_END_US         40
 #define EVENT_JITTER_US               16
+
+#define EVENT_RX_JITTER_US(phy) 16    /* Radio Rx timing uncertainty */
+#define EVENT_RX_TO_US(phy) (((((phy)&0x03) + 4)*8 / BIT((((phy)&0x3)>>1))) + \
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_vendor.h]: replace *8  with <<3

**Review by cvinayak — CHANGES_REQUESTED** [review]
one minor comment.

**Review by cvinayak — APPROVED** [review]

**Comment by cvinayak** [comment]
@carlescufi this is ready for merge.

**Comment by aescolar** [comment]
@carlescufi This PR is ready, could you please merge

**Pull request merged**